xplique/commons/operators.py

```diff
@@ -111,6 +111,102 @@ def segmentation_operator(model: Callable,
     scores = tf.reduce_sum(model(inputs) * targets, axis=(1, 2, 3))
     return scores
 
+class Tasks(Enum):
+    """
+    Enumeration of different tasks for which we have defined operators
+    """
+    CLASSIFICATION = predictions_operator
+    REGRESSION = regression_operator
+
+    @staticmethod
+    def from_string(operator_name: str) -> "Tasks":
+        """
+        Restore an operator from a string
+
+        Parameters
+        ----------
+        operator_name
+            String indicating the operator to restore: must be one
+            of 'classification' or 'regression'
+
+        Returns
+        -------
+        operator
+            The Tasks object
+        """
+        assert operator_name in [
+            "classification",
+            "regression",
+        ], "Only 'classification' and 'regression' are supported."
+
+        if operator_name == "regression":
+            return Tasks.REGRESSION
+        return Tasks.CLASSIFICATION
+
+def check_operator(operator: Callable):
+    """
+    Check if the operator is valid g(f, x, y) -> tf.Tensor
+    and raise an exception and return true if so.
+
+    Parameters
+    ----------
+    operator
+        Operator to check
+
+    Returns
+    -------
+    is_valid
+        True if the operator is valid, False otherwise.
+    """
+    # handle tf functions
+    # pylint: disable=protected-access
+    if hasattr(operator, '_python_function'):
+        return check_operator(operator._python_function)
+
+    # the operator must be callable
+    # pylint: disable=isinstance-second-argument-not-valid-type
+    if not isinstance(operator, Callable):
```

Comment:
The pylint warning makes sense for me. I suggest to change `not isinstance(operator, Callable)` by `not hastattr(operator, '__call__')`.

Reply:
Nice